apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -68,12 +68,7 @@ pub(crate) async fn handle_responses<T: HttpBody>(
                     })?;
 
                     let mut res_data = {
-                        // TODO: caching of the transformed JSONSelection with the selection set applied?
```

Comment:
I was thinking more broadly with this comment, that there's opportunity here to cache the transformed selection across GraphQL queries. Queries with the same selection set (which already implements Hash) could reuse the cached JSONSelection. So there might be more that can be done here, but that's outside the scope of this PR.

Reply:
agreed, i'll leave a comment on the existing issue about how this improvement is related but not the same